[Bug]: unable to install latest pybamm on aarch64

### PyBaMM Version

Pybamm 25.1.1

### Python Version

python 3.10

### Describe the bug

since solvers moved to new package pybammsolvers, i can not install pybamm on aarch64, even install pybammsolvers directly:

 pip install pybammsolvers
Defaulting to user installation because normal site-packages is not writeable
Collecting pybammsolvers
  Using cached pybammsolvers-0.1.0.tar.gz (44 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting casadi (from pybammsolvers)
  Using cached casadi-3.6.7-cp310-none-manylinux2014_aarch64.whl.metadata (2.2 kB)
Collecting numpy<2.0 (from pybammsolvers)
  Using cached numpy-1.26.4-cp310-cp310-manylinux_2_17_aarch64.manylinux2014_aarch64.whl.metadata (62 kB)
Using cached numpy-1.26.4-cp310-cp310-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (14.2 MB)
Using cached casadi-3.6.7-cp310-none-manylinux2014_aarch64.whl (43.9 MB)
Building wheels for collected packages: pybammsolvers
  Building wheel for pybammsolvers (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for pybammsolvers (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [321 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build/lib.linux-aarch64-cpython-310/pybammsolvers
      copying src/pybammsolvers/version.py -> build/lib.linux-aarch64-cpython-310/pybammsolvers
      copying src/pybammsolvers/__init__.py -> build/lib.linux-aarch64-cpython-310/pybammsolvers
      running egg_info
      writing src/pybammsolvers.egg-info/PKG-INFO
      writing dependency_links to src/pybammsolvers.egg-info/dependency_links.txt
      writing requirements to src/pybammsolvers.egg-info/requires.txt
      writing top-level names to src/pybammsolvers.egg-info/top_level.txt
      reading manifest file 'src/pybammsolvers.egg-info/SOURCES.txt'
      adding license file 'LICENSE'
      writing manifest file 'src/pybammsolvers.egg-info/SOURCES.txt'
      /tmp/pip-build-env-fbr81j_m/overlay/lib/python3.10/site-packages/setuptools/command/build_py.py:212: _Warning: Package 'pybammsolvers.idaklu_source' is absent from the `packages` configuration.
      !!
      
              ********************************************************************************
              ############################
              # Package would be ignored #
              ############################
              Python recognizes 'pybammsolvers.idaklu_source' as an importable package[^1],
              but it is absent from setuptools' `packages` configuration.
      
              This leads to an ambiguous overall configuration. If you want to distribute this
              package, please make sure that 'pybammsolvers.idaklu_source' is explicitly added
              to the `packages` configuration field.
      
              Alternatively, you can also rely on setuptools' discovery methods
              (for example by using `find_namespace_packages(...)`/`find_namespace:`
              instead of `find_packages(...)`/`find:`).
      
              You can read more about "package discovery" on setuptools documentation page:
      
              - https://setuptools.pypa.io/en/latest/userguide/package_discovery.html
      
              If you don't want 'pybammsolvers.idaklu_source' to be distributed and are
              already explicitly excluding 'pybammsolvers.idaklu_source' via
              `find_namespace_packages(...)/find_namespace` or `find_packages(...)/find`,
              you can try to use `exclude_package_data`, or `include-package-data=False` in
              combination with a more fine grained `package-data` configuration.
      
              You can read more about "package data files" on setuptools documentation page:
      
              - https://setuptools.pypa.io/en/latest/userguide/datafiles.html
      
      
              [^1]: For Python, any directory (with suitable naming) can be imported,
                    even if it does not contain any `.py` files.
                    On the other hand, currently there is no concept of package data
                    directory, all directories are treated like packages.
              ********************************************************************************
      
      !!
        check.warn(importable)
      /tmp/pip-build-env-fbr81j_m/overlay/lib/python3.10/site-packages/setuptools/command/build_py.py:212: _Warning: Package 'pybammsolvers.idaklu_source.Expressions' is absent from the `packages` configuration.
      !!
      
              ********************************************************************************
              ############################
              # Package would be ignored #
              ############################
              Python recognizes 'pybammsolvers.idaklu_source.Expressions' as an importable package[^1],
              but it is absent from setuptools' `packages` configuration.
      
              This leads to an ambiguous overall configuration. If you want to distribute this
              package, please make sure that 'pybammsolvers.idaklu_source.Expressions' is explicitly added
              to the `packages` configuration field.
      
              Alternatively, you can also rely on setuptools' discovery methods
              (for example by using `find_namespace_packages(...)`/`find_namespace:`
              instead of `find_packages(...)`/`find:`).
      
              You can read more about "package discovery" on setuptools documentation page:
      
              - https://setuptools.pypa.io/en/latest/userguide/package_discovery.html
      
              If you don't want 'pybammsolvers.idaklu_source.Expressions' to be distributed and are
              already explicitly excluding 'pybammsolvers.idaklu_source.Expressions' via
              `find_namespace_packages(...)/find_namespace` or `find_packages(...)/find`,
              you can try to use `exclude_package_data`, or `include-package-data=False` in
              combination with a more fine grained `package-data` configuration.
      
              You can read more about "package data files" on setuptools documentation page:
      
              - https://setuptools.pypa.io/en/latest/userguide/datafiles.html
      
      
              [^1]: For Python, any directory (with suitable naming) can be imported,
                    even if it does not contain any `.py` files.
                    On the other hand, currently there is no concept of package data
                    directory, all directories are treated like packages.
              ********************************************************************************
      
      !!
        check.warn(importable)
      /tmp/pip-build-env-fbr81j_m/overlay/lib/python3.10/site-packages/setuptools/command/build_py.py:212: _Warning: Package 'pybammsolvers.idaklu_source.Expressions.Base' is absent from the `packages` configuration.
      !!
      
              ********************************************************************************
              ############################
              # Package would be ignored #
              ############################
              Python recognizes 'pybammsolvers.idaklu_source.Expressions.Base' as an importable package[^1],
              but it is absent from setuptools' `packages` configuration.
      
              This leads to an ambiguous overall configuration. If you want to distribute this
              package, please make sure that 'pybammsolvers.idaklu_source.Expressions.Base' is explicitly added
              to the `packages` configuration field.
      
              Alternatively, you can also rely on setuptools' discovery methods
              (for example by using `find_namespace_packages(...)`/`find_namespace:`
              instead of `find_packages(...)`/`find:`).
      
              You can read more about "package discovery" on setuptools documentation page:
      
              - https://setuptools.pypa.io/en/latest/userguide/package_discovery.html
      
              If you don't want 'pybammsolvers.idaklu_source.Expressions.Base' to be distributed and are
              already explicitly excluding 'pybammsolvers.idaklu_source.Expressions.Base' via
              `find_namespace_packages(...)/find_namespace` or `find_packages(...)/find`,
              you can try to use `exclude_package_data`, or `include-package-data=False` in
              combination with a more fine grained `package-data` configuration.
      
              You can read more about "package data files" on setuptools documentation page:
      
              - https://setuptools.pypa.io/en/latest/userguide/datafiles.html
      
      
              [^1]: For Python, any directory (with suitable naming) can be imported,
                    even if it does not contain any `.py` files.
                    On the other hand, currently there is no concept of package data
                    directory, all directories are treated like packages.
              ********************************************************************************
      
      !!
        check.warn(importable)
      /tmp/pip-build-env-fbr81j_m/overlay/lib/python3.10/site-packages/setuptools/command/build_py.py:212: _Warning: Package 'pybammsolvers.idaklu_source.Expressions.Casadi' is absent from the `packages` configuration.
      !!
      
              ********************************************************************************
              ############################
              # Package would be ignored #
              ############################
              Python recognizes 'pybammsolvers.idaklu_source.Expressions.Casadi' as an importable package[^1],
              but it is absent from setuptools' `packages` configuration.
      
              This leads to an ambiguous overall configuration. If you want to distribute this
              package, please make sure that 'pybammsolvers.idaklu_source.Expressions.Casadi' is explicitly added
              to the `packages` configuration field.
      
              Alternatively, you can also rely on setuptools' discovery methods
              (for example by using `find_namespace_packages(...)`/`find_namespace:`
              instead of `find_packages(...)`/`find:`).
      
              You can read more about "package discovery" on setuptools documentation page:
      
              - https://setuptools.pypa.io/en/latest/userguide/package_discovery.html
      
              If you don't want 'pybammsolvers.idaklu_source.Expressions.Casadi' to be distributed and are
              already explicitly excluding 'pybammsolvers.idaklu_source.Expressions.Casadi' via
              `find_namespace_packages(...)/find_namespace` or `find_packages(...)/find`,
              you can try to use `exclude_package_data`, or `include-package-data=False` in
              combination with a more fine grained `package-data` configuration.
      
              You can read more about "package data files" on setuptools documentation page:
      
              - https://setuptools.pypa.io/en/latest/userguide/datafiles.html
      
      
              [^1]: For Python, any directory (with suitable naming) can be imported,
                    even if it does not contain any `.py` files.
                    On the other hand, currently there is no concept of package data
                    directory, all directories are treated like packages.
              ********************************************************************************
      
      !!
        check.warn(importable)
      /tmp/pip-build-env-fbr81j_m/overlay/lib/python3.10/site-packages/setuptools/command/build_py.py:212: _Warning: Package 'pybammsolvers.idaklu_source.Expressions.IREE' is absent from the `packages` configuration.
      !!
      
              ********************************************************************************
              ############################
              # Package would be ignored #
              ############################
              Python recognizes 'pybammsolvers.idaklu_source.Expressions.IREE' as an importable package[^1],
              but it is absent from setuptools' `packages` configuration.
      
              This leads to an ambiguous overall configuration. If you want to distribute this
              package, please make sure that 'pybammsolvers.idaklu_source.Expressions.IREE' is explicitly added
              to the `packages` configuration field.
      
              Alternatively, you can also rely on setuptools' discovery methods
              (for example by using `find_namespace_packages(...)`/`find_namespace:`
              instead of `find_packages(...)`/`find:`).
      
              You can read more about "package discovery" on setuptools documentation page:
      
              - https://setuptools.pypa.io/en/latest/userguide/package_discovery.html
      
              If you don't want 'pybammsolvers.idaklu_source.Expressions.IREE' to be distributed and are
              already explicitly excluding 'pybammsolvers.idaklu_source.Expressions.IREE' via
              `find_namespace_packages(...)/find_namespace` or `find_packages(...)/find`,
              you can try to use `exclude_package_data`, or `include-package-data=False` in
              combination with a more fine grained `package-data` configuration.
      
              You can read more about "package data files" on setuptools documentation page:
      
              - https://setuptools.pypa.io/en/latest/userguide/datafiles.html
      
      
              [^1]: For Python, any directory (with suitable naming) can be imported,
                    even if it does not contain any `.py` files.
                    On the other hand, currently there is no concept of package data
                    directory, all directories are treated like packages.
              ********************************************************************************
      
      !!
        check.warn(importable)
      copying src/pybammsolvers/idaklu.cpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers
      creating build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source
      copying src/pybammsolvers/idaklu_source/IDAKLUSolver.cpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source
      copying src/pybammsolvers/idaklu_source/IDAKLUSolver.hpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source
      copying src/pybammsolvers/idaklu_source/IDAKLUSolverGroup.cpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source
      copying src/pybammsolvers/idaklu_source/IDAKLUSolverGroup.hpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source
      copying src/pybammsolvers/idaklu_source/IDAKLUSolverOpenMP.hpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source
      copying src/pybammsolvers/idaklu_source/IDAKLUSolverOpenMP.inl -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source
      copying src/pybammsolvers/idaklu_source/IDAKLUSolverOpenMP_solvers.cpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source
      copying src/pybammsolvers/idaklu_source/IDAKLUSolverOpenMP_solvers.hpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source
      copying src/pybammsolvers/idaklu_source/IdakluJax.cpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source
      copying src/pybammsolvers/idaklu_source/IdakluJax.hpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source
      copying src/pybammsolvers/idaklu_source/Options.cpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source
      copying src/pybammsolvers/idaklu_source/Options.hpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source
      copying src/pybammsolvers/idaklu_source/Solution.cpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source
      copying src/pybammsolvers/idaklu_source/Solution.hpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source
      copying src/pybammsolvers/idaklu_source/SolutionData.cpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source
      copying src/pybammsolvers/idaklu_source/SolutionData.hpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source
      copying src/pybammsolvers/idaklu_source/common.cpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source
      copying src/pybammsolvers/idaklu_source/common.hpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source
      copying src/pybammsolvers/idaklu_source/idaklu_solver.hpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source
      copying src/pybammsolvers/idaklu_source/observe.cpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source
      copying src/pybammsolvers/idaklu_source/observe.hpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source
      copying src/pybammsolvers/idaklu_source/sundials_functions.hpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source
      copying src/pybammsolvers/idaklu_source/sundials_functions.inl -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source
      creating build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source/Expressions
      copying src/pybammsolvers/idaklu_source/Expressions/Expressions.hpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source/Expressions
      creating build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source/Expressions/Base
      copying src/pybammsolvers/idaklu_source/Expressions/Base/Expression.hpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source/Expressions/Base
      copying src/pybammsolvers/idaklu_source/Expressions/Base/ExpressionSet.hpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source/Expressions/Base
      copying src/pybammsolvers/idaklu_source/Expressions/Base/ExpressionTypes.hpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source/Expressions/Base
      creating build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source/Expressions/Casadi
      copying src/pybammsolvers/idaklu_source/Expressions/Casadi/CasadiFunctions.cpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source/Expressions/Casadi
      copying src/pybammsolvers/idaklu_source/Expressions/Casadi/CasadiFunctions.hpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source/Expressions/Casadi
      creating build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source/Expressions/IREE
      copying src/pybammsolvers/idaklu_source/Expressions/IREE/IREEBaseFunction.hpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source/Expressions/IREE
      copying src/pybammsolvers/idaklu_source/Expressions/IREE/IREEFunction.hpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source/Expressions/IREE
      copying src/pybammsolvers/idaklu_source/Expressions/IREE/IREEFunctions.cpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source/Expressions/IREE
      copying src/pybammsolvers/idaklu_source/Expressions/IREE/IREEFunctions.hpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source/Expressions/IREE
      copying src/pybammsolvers/idaklu_source/Expressions/IREE/ModuleParser.cpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source/Expressions/IREE
      copying src/pybammsolvers/idaklu_source/Expressions/IREE/ModuleParser.hpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source/Expressions/IREE
      copying src/pybammsolvers/idaklu_source/Expressions/IREE/iree_jit.cpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source/Expressions/IREE
      copying src/pybammsolvers/idaklu_source/Expressions/IREE/iree_jit.hpp -> build/lib.linux-aarch64-cpython-310/pybammsolvers/idaklu_source/Expressions/IREE
      running build_ext
      CMake Warning:
        Ignoring extra path from command line:
      
         "/tmp/pip-install-ml11b504/pybammsolvers_a88d40bc6d0547688960ec5d0c66dd00"
      
      
      CMake Error: The source directory "/tmp/pip-install-ml11b504/pybammsolvers_a88d40bc6d0547688960ec5d0c66dd00" does not appear to contain CMakeLists.txt.
      Specify --help for usage, or press the help button on the CMake GUI.
      ---------- Running CMake for IDAKLU solver ----------------------------------------
      Traceback (most recent call last):
        File "/home/liuchengkun/.local/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 389, in <module>
          main()
        File "/home/liuchengkun/.local/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 373, in main
          json_out["return_val"] = hook(**hook_input["kwargs"])
        File "/home/liuchengkun/.local/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 280, in build_wheel
          return _build_backend().build_wheel(
        File "/tmp/pip-build-env-fbr81j_m/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 438, in build_wheel
          return _build(['bdist_wheel', '--dist-info-dir', str(metadata_directory)])
        File "/tmp/pip-build-env-fbr81j_m/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 426, in _build
          return self._build_with_temp_dir(
        File "/tmp/pip-build-env-fbr81j_m/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 407, in _build_with_temp_dir
          self.run_setup()
        File "/tmp/pip-build-env-fbr81j_m/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 320, in run_setup
          exec(code, locals())
        File "<string>", line 285, in <module>
        File "/tmp/pip-build-env-fbr81j_m/overlay/lib/python3.10/site-packages/setuptools/__init__.py", line 117, in setup
          return distutils.core.setup(**attrs)
        File "/tmp/pip-build-env-fbr81j_m/overlay/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 186, in setup
          return run_commands(dist)
        File "/tmp/pip-build-env-fbr81j_m/overlay/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 202, in run_commands
          dist.run_commands()
        File "/tmp/pip-build-env-fbr81j_m/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 983, in run_commands
          self.run_command(cmd)
        File "/tmp/pip-build-env-fbr81j_m/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 999, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-fbr81j_m/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 1002, in run_command
          cmd_obj.run()
        File "<string>", line 232, in run
        File "/tmp/pip-build-env-fbr81j_m/overlay/lib/python3.10/site-packages/setuptools/command/bdist_wheel.py", line 379, in run
          self.run_command("build")
        File "/tmp/pip-build-env-fbr81j_m/overlay/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 339, in run_command
          self.distribution.run_command(command)
        File "/tmp/pip-build-env-fbr81j_m/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 999, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-fbr81j_m/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 1002, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-fbr81j_m/overlay/lib/python3.10/site-packages/setuptools/_distutils/command/build.py", line 136, in run
          self.run_command(cmd_name)
        File "/tmp/pip-build-env-fbr81j_m/overlay/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 339, in run_command
          self.distribution.run_command(command)
        File "/tmp/pip-build-env-fbr81j_m/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 999, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-fbr81j_m/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 1002, in run_command
          cmd_obj.run()
        File "<string>", line 138, in run
        File "/usr/local/lib/python3.10/subprocess.py", line 526, in run
          raise CalledProcessError(retcode, process.args,
      subprocess.CalledProcessError: Command '['cmake', '/tmp/pip-install-ml11b504/pybammsolvers_a88d40bc6d0547688960ec5d0c66dd00', '-DCMAKE_BUILD_TYPE=RELEASE', '-DPYTHON_EXECUTABLE=/usr/local/bin/python3.10', '-DUSE_PYTHON_CASADI=TRUE', '-DPYBAMM_IDAKLU_EXPR_CASADI=ON', '-DPYBAMM_IDAKLU_EXPR_IREE=OFF', '-DSuiteSparse_ROOT=/tmp/pip-install-ml11b504/pybammsolvers_a88d40bc6d0547688960ec5d0c66dd00/.idaklu', '-DSUNDIALS_ROOT=/tmp/pip-install-ml11b504/pybammsolvers_a88d40bc6d0547688960ec5d0c66dd00/.idaklu']' returned non-zero exit status 1.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pybammsolvers
Failed to build pybammsolvers
ERROR: Failed to build installable wheels for some pyproject.toml based projects (pybammsolvers)

### Steps to Reproduce

pip install pybammsolvers

### Relevant log output

```shell

```

--- a/.github/workflows/_build_solver_wheels.yml
+++ b/.github/workflows/_build_solver_wheels.yml
@@ -142,11 +142,33 @@ jobs:
           if-no-files-found: error
 
   build_manylinux_wheels:
-    name: Wheels (Linux)
+    name: Wheels (Linux ${{ matrix.arch }})
     if: ${{ contains(fromJSON(inputs.platforms), 'linux') }}
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.runner }}
     permissions:
       contents: read
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          # x86_64 uses the default manylinux2014; CasADi's x86_64 wheel is
+          # built with the legacy libstdc++ ABI (CentOS 7, EPEL disabled).
+          - arch: x86_64
+            runner: ubuntu-latest
+            artifact: wheels_manylinux
+            before_all: >
+              yum -y --disablerepo=epel install openblas-devel lapack-devel &&
+              python install_KLU_Sundials.py
+          # aarch64 needs manylinux_2_34: CasADi's aarch64 wheel is built with
+          # the C++11 ABI and requires a GCC>=11 libstdc++, which manylinux2014
+          # and manylinux_2_28 lack. openblas/lapack live in AlmaLinux 9's CRB.
+          - arch: aarch64
+            runner: ubuntu-24.04-arm
+            artifact: wheels_manylinux_aarch64
+            before_all: >
+              dnf config-manager --set-enabled crb &&
+              dnf -y install openblas-devel lapack-devel &&
+              python install_KLU_Sundials.py
     steps:
       - uses: actions/checkout@9c091bb21b7c1c1d1991bb908d89e4e9dddfe3e0 # v7.0.0
         with:
@@ -169,18 +191,17 @@ jobs:
         uses: actions/cache@55cc8345863c7cc4c66a329aec7e433d2d1c52a9 # v6.1.0
         with:
           path: packages/pybammsolvers/wheelhouse
-          key: solver-wheels-linux-${{ steps.solverhash.outputs.hash }}
+          key: solver-wheels-linux-${{ matrix.arch }}-${{ steps.solverhash.outputs.hash }}
 
       - name: Build wheels on Linux
         if: ${{ !(inputs.cache_wheels && steps.cache.outputs.cache-hit == 'true') }}
         run: pipx run cibuildwheel --output-dir wheelhouse
         env:
-          CIBW_ARCHS_LINUX: x86_64
-          # openblas-devel/lapack-devel live in PowerTools, not EPEL; disable EPEL
-          # so its periodic metadata-rotation outages can't break the wheel build.
-          CIBW_BEFORE_ALL_LINUX: >
-            yum -y --disablerepo=epel install openblas-devel lapack-devel &&
-            python install_KLU_Sundials.py
+          CIBW_ARCHS_LINUX: ${{ matrix.arch }}
+          # aarch64 requires GCC>=11 libstdc++ for CasADi's C++11-ABI wheel;
+          # ignored for the x86_64 build (which uses the default manylinux2014).
+          CIBW_MANYLINUX_AARCH64_IMAGE: manylinux_2_34
+          CIBW_BEFORE_ALL_LINUX: ${{ matrix.before_all }}
           # libcasadi ships as a pip dep (resolved at runtime via $ORIGIN/../casadi).
           CIBW_REPAIR_WHEEL_COMMAND_LINUX: auditwheel repair --exclude libcasadi.so.3.7 -w {dest_dir} {wheel}
           CIBW_TEST_EXTRAS: "dev"
@@ -191,7 +212,7 @@ jobs:
       - name: Upload wheels for Linux
         uses: actions/upload-artifact@043fb46d1a93c77aae656e7c1c64a875d1fc6a0a # v7.0.1
         with:
-          name: wheels_manylinux
+          name: ${{ matrix.artifact }}
           path: packages/pybammsolvers/wheelhouse/*.whl
           if-no-files-found: error
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,7 @@
 
 ## Features
 
+- `pybammsolvers` now ships Linux `aarch64` (arm64) wheels, built on `manylinux_2_34` native ARM runners. The libstdc++ `std::string` ABI (`_GLIBCXX_USE_CXX11_ABI`) is now detected from the linked CasADi library instead of hard-coded, resolving the unresolved-symbol failure against CasADi's C++11-ABI aarch64 wheel. ([#5653](https://github.com/pybamm-team/PyBaMM/pull/5653))
 - Added `skip_surface_form_check` option to `EISSimulation` to bypass the surface form validation. ([#5632](https://github.com/pybamm-team/PyBaMM/pull/5632))
 - Added ability to export pybamm.Simulation with experiments to DiffSL format ([#5557)](https://github.com/pybamm-team/PyBaMM/pull/5557))
 - PyBaMM and `pybammsolvers` now develop in a single repository — a UV workspace under `packages/` — while continuing to release independently to PyPI. Release tags are namespaced (`pybamm-v*` and `pybammsolvers-v*`), and PyBaMM's CI now tests against the in-repo solver on every platform. The published `pybamm` package and its dependency on `pybammsolvers` are unchanged for users. See `RELEASE.md` for the release model. ([#5512](https://github.com/pybamm-team/PyBaMM/issues/5512))

--- a/packages/pybammsolvers/CMakeLists.txt
+++ b/packages/pybammsolvers/CMakeLists.txt
@@ -76,8 +76,8 @@ if (NOT MSVC)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror=vla")
 endif ()
 
-# casadi seems to compile without the newer versions of std::string
-add_compile_definitions(_GLIBCXX_USE_CXX11_ABI=0)
+# The libstdc++ std::string ABI (_GLIBCXX_USE_CXX11_ABI) is reconciled against
+# the actual libcasadi build further down, once CasADi has been located.
 
 find_package(pybind11 CONFIG REQUIRED)
 
@@ -222,6 +222,46 @@ else ()
     else()
         target_link_libraries(idaklu PRIVATE casadi)
     endif()
+endif ()
+
+# Match idaklu's libstdc++ std::string ABI to the linked libcasadi (GNU only;
+# no-op for MSVC and macOS/libc++). CasADi's PyPI casadi-config.cmake forces
+# `_GLIBCXX_USE_CXX11_ABI=0` on consumers: correct for the x86_64 wheel (legacy
+# ABI) but wrong for the aarch64 wheel (C++11 ABI), leaving casadi's std::string
+# symbols unresolved. Detect libcasadi's real ABI and override CasADi's guess.
+if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    # Drop CasADi's hard-coded (and, on aarch64, wrong) ABI interface definition.
+    if (TARGET casadi::casadi)
+        get_target_property(_casadi_defs casadi::casadi INTERFACE_COMPILE_DEFINITIONS)
+        if (_casadi_defs)
+            list(REMOVE_ITEM _casadi_defs
+                "_GLIBCXX_USE_CXX11_ABI=0" "_GLIBCXX_USE_CXX11_ABI=1")
+            set_target_properties(casadi::casadi PROPERTIES
+                INTERFACE_COMPILE_DEFINITIONS "${_casadi_defs}")
+        endif ()
+    endif ()
+
+    # Key off casadi::Function::deserialize(const std::string&): the C++11-ABI
+    # build exports the __cxx11 mangling, the legacy build the `Ss` mangling.
+    # Default to the legacy ABI if the probe cannot run (never worse than before).
+    set(_casadi_cxx11_abi 0)
+    set(_nm "${CMAKE_NM}")
+    if (NOT _nm)
+        find_program(_nm NAMES nm gcc-nm)
+    endif ()
+    file(GLOB _libcasadi "${CASADI_LIB_DIR}/libcasadi.so*")
+    if (_libcasadi AND _nm)
+        list(GET _libcasadi 0 _libcasadi0)
+        execute_process(
+            COMMAND "${_nm}" -D --defined-only "${_libcasadi0}"
+            OUTPUT_VARIABLE _casadi_syms
+            ERROR_QUIET)
+        if (_casadi_syms MATCHES "Function11deserializeERKNSt7__cxx11")
+            set(_casadi_cxx11_abi 1)
+        endif ()
+    endif ()
+    message(STATUS "idaklu: _GLIBCXX_USE_CXX11_ABI=${_casadi_cxx11_abi} (matched to libcasadi)")
+    target_compile_definitions(idaklu PRIVATE _GLIBCXX_USE_CXX11_ABI=${_casadi_cxx11_abi})
 endif ()
 
 # Math flags for the IDAKLU target.

